deployment-tinkerer/src/main/java/org/wso2/testgrid/deployment/tinkerer/utils/SSHHelper.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.testgrid.deployment.tinkerer.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.Agent;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Scanner;
+import java.util.Set;
+
+/**
+ * This class contains the SSH config file operations.
+ * The file is located at the path "~/.ssh/config"
+ *
+ * @since 1.0.0
+ */
+public class SSHHelper {
+
+    private static final Logger logger = LoggerFactory.getLogger(SSHHelper.class);
+
+    /**
+     *
+     * Adds and entry to the ssh config file  that will enable direct access to the instance
+     * via the bastian instance.
+     *
+     * The instance will contain a proxy access entry that will route the connection via the bastian
+     * instance.
+     * @param keyContent the content of the security key file (.pem file) that is used to access the instance
+     * @param bastianIP The IP address of the bastian instance.
+     * @param agent The {@link Agent} object corresponding to the remote agent running on the instance
+     * @throws IOException When there is an error creating the config entry
+     */
+    public static void addConfigEntry(String keyContent, String bastianIP, Agent agent) throws IOException {
+
+        Path config = Paths.get(System.getProperty("user.home"), ".ssh", "config");
+        Path keyPath = Paths.get(System.getProperty("user.home"), ".ssh", agent.getInstanceId() + "-key.pem");
```

Comment:
As a general rule, it's good if we do not touch system/user config files that may have its own lifecycle than the LC of the test-plan. The ~/.ssh/config is a user configuration file. 

IMO, we should create this file within the test-plan's workspace, and provide it as an input into SSH command. You can use _[-F](https://www.ssh.com/ssh/command/)_ cmd argument for that purpose. Shall we create a new ticket to track this?

Reply:
.